analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -64,24 +69,22 @@ private static ProgramState LearnBranchingConstraint<T>(ProgramState state, IBin
 
     private static SymbolicConstraint BinaryConstraint(BinaryOperatorKind kind, SymbolicValue left, SymbolicValue right)
     {
-        if (left is null && right is null)
-        {
-            return null;
-        }
-        else if (left is null || right is null)
+        var leftBool = left?.Constraint<BoolConstraint>();
+        var rightBool = right?.Constraint<BoolConstraint>();
+        if (leftBool is not null ^ rightBool is not null)
```

Comment:
This is the same, isn't it? I find the `is null` easier to follow (positive logic)
```suggestion
        if (leftBool is null ^ rightBool is null)
```

Reply:
Yep. I think so. Tim was the first one who came up with xor for this kind of test on another PR, so we may ask him ;-).